apollo-client/src/main/java/com/ctrip/framework/apollo/monitor/internal/listener/impl/DefaultApolloClientThreadPoolApi.java

```diff
@@ -0,0 +1,177 @@
+  public static final String ABSTRACT_CONFIG_FILE = AbstractConfigFile.class.getSimpleName();
+  public static final String METRICS_EXPORTER = AbstractApolloClientMetricsExporter.class.getSimpleName();
+  private final Map<String, ApolloThreadPoolInfo> executorMap = Maps.newHashMap();
+
+  public DefaultApolloClientThreadPoolApi(
+      ExecutorService remoteConfigRepositoryExecutorService,
+      ExecutorService abstractConfigExecutorService,
+      ExecutorService abstractConfigFileExecutorService,
+      ExecutorService metricsExporterExecutorService) {
+    super(TAG_THREAD_POOL);
+    executorMap.put(REMOTE_CONFIG_REPOSITORY,
+        new ApolloThreadPoolInfo((ThreadPoolExecutor) remoteConfigRepositoryExecutorService));
+    executorMap.put(ABSTRACT_CONFIG,
+        new ApolloThreadPoolInfo((ThreadPoolExecutor) abstractConfigExecutorService));
+    executorMap.put(ABSTRACT_CONFIG_FILE,
+        new ApolloThreadPoolInfo((ThreadPoolExecutor) abstractConfigFileExecutorService));
+    executorMap.put(METRICS_EXPORTER,
+        new ApolloThreadPoolInfo((ThreadPoolExecutor) metricsExporterExecutorService));
+  }
+
+  @Override
+  public void export0() {
+    executorMap.forEach((key, value) -> exportThreadPoolMetrics(value, key));
+  }
+
+  private void exportThreadPoolMetrics(ApolloThreadPoolInfo info, String name) {
+    List<Double> metrics = Arrays.asList(
+        (double) info.getActiveTaskCount(),
+        (double) info.getQueueSize(),
+        (double) info.getCompletedTaskCount(),
+        (double) info.getPoolSize(),
+        (double) info.getTotalTaskCount(),
+        (double) info.getCorePoolSize(),
+        (double) info.getMaximumPoolSize(),
+        (double) info.getLargestPoolSize(),
+        (double) info.getQueueCapacity(),
+        (double) info.getQueueRemainingCapacity(),
+        info.getCurrentLoad()
+    );
+
+    for (int i = 0; i < metrics.size(); i++) {
+      String key = name + METRICS_THREAD_POOL_PARAMS[i + 1];
+      createOrUpdateGaugeSample(key, METRICS_THREAD_POOL_PARAMS[i + 1],
+          Collections.singletonMap(METRICS_THREAD_POOL_PARAMS[0], name), metrics.get(i));
+    }
+  }
+
+
+  @Override
+  public boolean isMetricsSampleUpdated() {
+    // memory status special
+    return true;
+  }
+
+  @Override
+  public Map<String, ApolloThreadPoolInfo> getThreadPoolInfo() {
+    return executorMap;
+  }
+
+  @Override
+  public ApolloThreadPoolInfo getRemoteConfigRepositoryThreadPoolInfo() {
+    return executorMap.get(REMOTE_CONFIG_REPOSITORY);
+  }
+
+  @Override
+  public ApolloThreadPoolInfo getAbstractConfigThreadPoolInfo() {
+    return executorMap.get(ABSTRACT_CONFIG);
+  }
+
+  @Override
+  public ApolloThreadPoolInfo getAbstractConfigFileThreadPoolInfo() {
+    return executorMap.get(ABSTRACT_CONFIG_FILE);
+  }
+
+  @Override
+  public ApolloThreadPoolInfo getMetricsExporterThreadPoolInfo() {
+    return executorMap.get(METRICS_EXPORTER);
+  }
+
+  public static class ApolloThreadPoolInfo {
```

Comment:
ApolloThreadPoolInfo  -> ApolloClientThreadPoolInfo 

and move it to Api's static class